common/src/main/java/org/bf2/common/OperandUtils.java

```diff
@@ -79,15 +88,74 @@ public static <T extends HasMetadata> T createOrUpdate(MixedOperation<T, ?, ?> r
         return withName.createOrReplace(resource);
     }
 
-    public static Affinity buildZookeeperPodAffinity(ManagedKafka managedKafka) {
+    /**
+     * Will likely always be a single toleration, but returns a list to gracefully handle null as empty
+     */
+    public static List<Toleration> profileTolerations(ManagedKafka managedKafka) {
+        String type =
+                OperandUtils.getOrDefault(managedKafka.getMetadata().getLabels(), ManagedKafka.PROFILE_TYPE, null);
+
+        if (type == null) {
+            return Collections.emptyList();
+        }
+
+        return Collections.singletonList(new TolerationBuilder()
+                .withKey(ManagedKafka.PROFILE_TYPE)
+                .withValue(type)
+                .withEffect("NoExecute")
+                .build());
+    }
+
+    public static NodeAffinity nodeAffinity(ManagedKafkaAgent agent, ManagedKafka managedKafka) {
+        if (agent == null || agent.getSpec().getCapacity().size() <= 1) {
+            return null; // not expected to use a node label
+        }
```

Comment:
Is the default machine pool used over the named machine pool in a single profile cluster? I thought if we use auto-scaler then we use named pools.

was also thinking would it be better to extract this code as an interface method.

Reply:
This check along with the code that deals with the profile being absent are mostly for legacy / backwards compatibility.  As we move forward these cases won't actually occur in staging / production - it will be required for the agent to be configured with the appropriate profiles and for those to be present on the managedkafka (should already be the case as evals are now gone).

Reply:
Thanks